### ISSUE TITLE
Unexpected LDAPSocketOpenError

### ISSUE THREAD

**Issue by swtch1** [cannatag/ldap3#507]
While running ldap3 v2.4.1 I am receiving the unexpected LDAPSocketOpenError below:
```
ldap3.core.exceptions.LDAPSocketOpenError: unable to send message, socket is not open
```
This error is unexpected because the same code works under v2.4.0.  Unfortunately I am unable to reproduce this error in a simple setting.  I tried many simple use cases with different sync strategies but everything there works, even calling the failing function directly.  Only when I run this under Flask and query my API do I see the error.  I reverted back to v2.4.0 and I don't see the issue.

Here are the last few lines of my stack trace:
```
 File "/usr/lib/python3.4/site-packages/doorman-1.4.2-py3.4.egg/doorman/core/asset/userorgroup.py", line 97, in _get_groups
    self._search_groups_members(groups),
  File "/usr/lib/python3.4/site-packages/doorman-1.4.2-py3.4.egg/doorman/core/asset/userorgroup.py", line 140, in _search_groups_members
    attributes=['member'],
  File "/usr/lib/python3.4/site-packages/ldap3/core/connection.py", line 775, in search
    response = self.post_send_search(self.send('searchRequest', request, controls))
  File "/usr/lib/python3.4/site-packages/ldap3/strategy/base.py", line 310, in send
    raise LDAPSocketOpenError(self.connection.last_error)
ldap3.core.exceptions.LDAPSocketOpenError: unable to send message, socket is not open
```

If this is not helpful this issue can be closed, but I wanted to add some validation for other users that may be seeing this.

Python Version: 3.4.5
Operating System: Centos 7.3.1611
Flask Version: 0.12.2

**Comment by cannatag** [comment]
Hi, you should try to check if the connection is open. Could you create a log file at debug level? Or simply put a print statement for each open, bind and unbind in your code?

**Comment by cannatag** [comment]
Hi, any news on this issue? I've just released the 2.5 version of ldap3. Can you give a try?

**Comment by swtch1** [comment]
This works on the latest version. Thank you.
